Tests/Adapter/SymfonyCacheTest.php

```diff
@@ -111,4 +111,147 @@ public function testCacheActionWithInvalidType()
 
         $this->cache->cacheAction('token', 'invalid-type');
     }
+
+    /**
+     * Asserts the flush method throws an exception if the IP version of the server cannot be detected.
+     */
+    public function testFlushThrowsExceptionWithWrongIP()
+    {
+        $cache = new SymfonyCache(
+            $this->router,
+            $this->filesystem,
+            '/cache/dir',
+            'token',
+            false,
+            array('all', 'translations'),
+            array(
+                array('ip' => 'wrong ip'),
+            )
+        );
+
+        $this->setExpectedException('\InvalidArgumentException', '"wrong ip" is not a valid ip address');
+
+        $cache->flush();
+    }
+
+    /**
+     * Tests the flush method with IPv4.
+     */
+    public function testFlushWithIPv4()
+    {
+        $mockBuilderClass = $this->getMockBuilderClass();
+
+        $cache = new SymfonyCache(
+            $this->router,
+            $this->filesystem,
+            '/cache/dir',
+            'token',
+            false,
+            array('all', 'translations'),
+            array(
+                array('ip' => '213.186.35.9', 'domain' => 'www.example.com', 'basic' => false, 'port' => 80),
+            )
+        );
+
+        $that = $this;
```

Comment:
If you think it's possible to replace this black magic with something like `Closure::bindTo()`, please add a `NEXT_MAJOR` note here too.